applications/Unity.GrantManager/src/Unity.GrantManager.HttpApi/Controllers/AttachmentController.cs

```diff
@@ -154,10 +158,55 @@ public async Task<IActionResult> DownloadChefsAttachment(Guid formSubmissionId,
             {
                 string ExceptionMessage = ex.Message;
                 logger.LogError(ex, "AttachmentController->DownloadChefsAttachment: {ExceptionMessage}", ExceptionMessage);
-                return StatusCode(500, "An error occurred while downloading the file.");
+                return StatusCode(500, errorFileMsg);
             }
         }
 
+        [HttpPost("chefs/download-all")]
+        [Consumes("application/json")]
+        public async Task<IActionResult> DownloadAllChefsAttachment([FromBody] List<AttachmentsDto> input)
+        {
+            var files = new List<FileContentResult>();
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            foreach (var item in input)

```

Comment:
We should discuss this in devops meeting, as this approach can cause a large amount of memory usage is a worst case scenario - e.g. 80 files and 25mb each will grow up to 2gb (or more based on the loops etc..)

Reply:
I agree, I’ll bring this up at our DevOps meeting. I’m really hoping that Chef’s API has an endpoint that lets us get all the files at once. However, they currently only have an endpoint that retrieves each file based on fileId, which a bit of hassle. James and I discussed about this downloadAll service but the challenge is getting them in memory. There’s another feature coming that includes checkboxes in the table, which I think will help maybe adding a limit on the number of files to be downloaded and compress into a zip file. I hope this should make things a lot easier for us. E.g  let users select a maximum of 5 files to zip and download

Reply:
@AndreGAot - Are you thinking of a streaming api controller where the S3 objects are coming in and being streamed (in-memory) to a zip file response object?

Reply:
@robinwn - not sure yet.. :) - need to do some digging and also see if we can figure out what a real world scenario really looks like for this scenario